main/params.go

```diff
@@ -419,138 +477,216 @@ func init() {
+
+		minValidatorStake := v.GetUint64(minValidatorStakeKey)
+		maxValidatorStake := v.GetUint64(maxValidatorStakeKey)
+		minDelegatorStake := v.GetUint64(minDelegatorStakeKey)
+		minDelegationFee := v.GetUint64(minDelegatorFeeKey)
+		if minValidatorStake > maxValidatorStake {
+			return errors.New("minimum validator stake can't be greater than maximum validator stake")
 		}
 
-		Config.MinValidatorStake = *minValidatorStake
-		Config.MaxValidatorStake = *maxValidatorStake
-		Config.MinDelegatorStake = *minDelegatorStake
+		Config.MinValidatorStake = minValidatorStake
+		Config.MaxValidatorStake = maxValidatorStake
+		Config.MinDelegatorStake = minDelegatorStake
 
-		if *minDelegationFee > 1000000 {
-			errs.Add(errors.New("delegation fee must be in the range [0, 1000000]"))
+		if minDelegationFee > 1000000 {
+			return errors.New("delegation fee must be in the range [0, 1000000]")
 		}
-		Config.MinDelegationFee = uint32(*minDelegationFee)
+		Config.MinDelegationFee = uint32(minDelegationFee)
 
 		if Config.MinStakeDuration == 0 {
-			errs.Add(errors.New("min stake duration can't be zero"))
+			return errors.New("min stake duration can't be zero")
 		}
 		if Config.MaxStakeDuration < Config.MinStakeDuration {
-			errs.Add(errors.New("max stake duration can't be less than min stake duration"))
+			return errors.New("max stake duration can't be less than min stake duration")
 		}
 		if Config.StakeMintingPeriod < Config.MaxStakeDuration {
-			errs.Add(errors.New("stake minting period can't be less than max stake duration"))
+			return errors.New("stake minting period can't be less than max stake duration")
 		}
 	} else {
 		Config.Params = *genesis.GetParams(networkID)
 	}
+
+	// Consensus Parameters
+	Config.ConsensusParams.K = v.GetInt(snowSampleSizeKey)
+	Config.ConsensusParams.Alpha = v.GetInt(snowQuorumSizeKey)
+	Config.ConsensusParams.BetaVirtuous = v.GetInt(snowVirtuousCommitThresholdKey)
+	Config.ConsensusParams.BetaRogue = v.GetInt(snowRogueCommitThresholdKey)
+	Config.ConsensusParams.Parents = v.GetInt(snowAvalancheNumParentsKey)
+	Config.ConsensusParams.BatchSize = v.GetInt(snowAvalancheBatchSizeKey)
+	Config.ConsensusParams.ConcurrentRepolls = v.GetInt(snowConcurrentRepollsKey)
+
+	Config.ConsensusGossipFrequency = v.GetDuration(consensusGossipFrequencyKey)
+	Config.ConsensusShutdownTimeout = v.GetDuration(consensusShutdownTimeoutKey)
+
+	// Assertions
+	Config.EnableAssertions = v.GetBool(assertionsEnabledKey)
+
+	// Crypto
+	Config.EnableCrypto = v.GetBool(signatureVerificationEnabledKey)
+
+	// Coreth Plugin
+	corethConfig, err := json.Marshal(v.Get(corethConfigKey))
+	if err != nil {
+		return fmt.Errorf("couldn't parse coreth config: %w", err)
+	}
+	Config.CorethConfig = string(corethConfig)
+
+	return nil
+}
+
+func parseViper() error {
+	v, err := getViper()
+	if err != nil {
+		return err
+	}
+
+	if v.GetBool(versionKey) {
+		format := "%s ["
+		args := []interface{}{
+			node.Version,
+		}
+
+		{
```

Comment:
Do we need this code to be in the `{` `}`  block? It works for me without them

Reply:
Removed